### PR TITLE
Do not allow users to login with display_name

### PR DESCRIPTION
**Pull request by camallen**
Removed ability to login with display_name, removed unique constraint on user#display_name index, switched to normal model validations for user#login and removed old code.

I still need to check that this won't break the front end but i'm almost 100% sure we don't use unique display_name filtering for anything anymore.

### CODE REVIEW

File: db/migrate/20151106172531_remove_uniqueness_display_name_index_users.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class RemoveUniquenessDisplayNameIndexUsers < ActiveRecord::Migration
+  def change
+    remove_index :users, column: :display_name, unique: true
```
Comment (marten) [inline db/migrate/20151106172531_remove_uniqueness_display_name_index_users.rb]: Dont we still want display names to be unique? What happens if I start impersonating moderators by setting my display name the same?

Comment (camallen) [inline db/migrate/20151106172531_remove_uniqueness_display_name_index_users.rb]: I'm not sure that it's a real problem as we report the @user  handle and the display name together in talk.

image

Comment (marten) [inline db/migrate/20151106172531_remove_uniqueness_display_name_index_users.rb]: Guess not then.

Comment (camallen) [inline db/migrate/20151106172531_remove_uniqueness_display_name_index_users.rb]: Can you think of another case where we'd need this? It'd be nice not to have to fix any conflicts if we remove this validation.

Comment (marten) [inline db/migrate/20151106172531_remove_uniqueness_display_name_index_users.rb]: I'm fine with multiple Jim's having the same display name. But we will always have to include the username too. Don't know of any other areas where we might not be doing that currently.

Comment (camallen) [inline db/migrate/20151106172531_remove_uniqueness_display_name_index_users.rb]: @brian-c @aweiksnar @parrish does the front end require unique display_names anything (filtering?) these days? Pretty sure the API uniquely filters on slugs and owners only.

**Comment by camallen** [comment]
I think we are good to go on this.

**Comment by marten** [comment]
LGTM

**Pull request merged**